src/Sylius/Behat/Context/Ui/Shop/LoginContext.php

```diff
@@ -236,6 +236,7 @@ public function iShouldBeNotifiedAboutDisabledAccount(): void
 
     /**
      * @Then I should be notified that email with reset instruction has been sent
+     * @Then I should be notified that the user email does not exist
```

Comment:
Why are there two conflicting `@Then` expectations here?

Reply:
It's the reason I ask @lchrusciel to know if we wanted to mirror the behaviour of the UI.
This conflicting line is here to demonstrate that there is an issue here.

I discussed for a long time with my marketing team about this sentence :

```
If the email you have specified exists in our system, we have sent there an instruction on how to reset your password.
```

and I have to add more explanations for our customers, because it's an OK message telling the email has been sent and some customers don't read the rest of the sentence.

We had loads of support tickets about this fact so if this is the behaviour of most customer portal, it's because there is a reason for that.

From my point of view, one of the reason could be a security one to avoid sending multiple time the same reset password email to the same customers or all of them (if the evil visitor know all accounts) and block the company because it will be flag as spammer for one of those reasons :

- to much email sent
- reported by users as spammers
- blocked by the email provider (internal policies for ex)
- rate limiter on the email server
- etc

Reply:
I don't think this affects the potential amount of unwanted emails in case of an attack attempt. If I make a 100 requests and submit 100 email addresses, the amount of undesired emails sent will match the number of successful email guesses regardless of whether or not the attacker receives success or failure responses.